webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3GroupHeader.scala

```diff
@@ -0,0 +1,42 @@
+package com.scalableminds.webknossos.datastore.datareaders.zarr3
+
+import com.scalableminds.webknossos.datastore.datareaders.zarr.NgffMetadataV2.jsonFormat
+import com.scalableminds.webknossos.datastore.datareaders.zarr.NgffMetadataV2
+import com.scalableminds.webknossos.datastore.helpers.JsonImplicits
+import play.api.libs.json._
+
+case class Zarr3GroupHeader(
+    zarr_format: Int, // must be 3
+    node_type: String, // must be "group"
+    ngffMetadata: Option[NgffMetadataV2],
+)
+
+object Zarr3GroupHeader extends JsonImplicits {
+
+  def FILENAME_ZARR_JSON = "zarr.json"
+  implicit object Zarr3GroupHeaderFormat extends Format[Zarr3GroupHeader] {
+    override def reads(json: JsValue): JsResult[Zarr3GroupHeader] =
+      for {
+        zarr_format <- (json \ "zarr_format").validate[Int]
+        node_type <- (json \ "node_type").validate[String]
+        ngffMetadata = (json \ "attributes" \ "ome").validate[NgffMetadataV2].asOpt
+      } yield
+        Zarr3GroupHeader(
+          zarr_format,
+          node_type,
+          ngffMetadata,
+        )
+
+    override def writes(zarr3GroupHeader: Zarr3GroupHeader): JsValue = {
+      val groupHeaderBuilder = Json.newBuilder
+      groupHeaderBuilder ++= Seq(
+        "zarr_format" -> zarr3GroupHeader.zarr_format,
+        "node_type" -> zarr3GroupHeader.node_type,
+      )
+      if (zarr3GroupHeader.ngffMetadata.isDefined) {
+        groupHeaderBuilder += ("attributes" -> Json.obj("ome" -> zarr3GroupHeader.ngffMetadata.get))
+      }
+      groupHeaderBuilder.result()
+    }
+  }
```

Comment:
is there a reason for the custom reads and writes? can the generated ones from Json.format not handle some of the data in here? If so, could you add a brief comment explaining it?

Reply:
ping

Reply:
The manual json serialization and deserialization is needed to have the correct json format. The default formatter would not put the metadata under "attribbutes.ome" but under "ngffMetadata" as thats how the member of `Zarr3GroupHeader` is called. Therefore the manual serialization is necessary here :) I added comments about this